src/Catalog/Registration/RegistrationPersistence.cs

```diff
@@ -155,9 +151,8 @@ private static async Task<IGraph> LoadCatalog(IStorage storage, Uri resourceUri,
 
         private static async Task<IGraph> LoadCatalogPage(IStorage storage, Uri pageUri, CancellationToken cancellationToken)
         {
-            Trace.TraceInformation("RegistrationPersistence.LoadCatalogPage: pageUri = {0}", pageUri);
             string json = await storage.LoadStringAsync(pageUri, cancellationToken);
-            IGraph graph = Utils.CreateGraph(json);
+            IGraph graph = Utils.CreateGraph(pageUri, json);
```

Comment:
Nit: return directly?

Reply:
I left this since I put a break point here a couple times while debugging to inspect the graph. This is the big guy that is eating memory.